src/components/Datagrid/types.ts

```diff
@@ -5,7 +5,9 @@ export interface AvailableColumn {
   title: string;
   width: number;
   group?: string;
-  metaGroup?: string; // Description for column picker
+  metaGroup?: string | null; // Description for column picker
+  // null means that column is not visible in column picker
```

Comment:
So I guess you're right this all can be handled just by not passing this field, because in static columns `metaGroup` changes nothing and in dynamic columns falsy means hidden